test/optimization/test_admm_miskp.py

```diff
@@ -1,199 +1,95 @@
-                    -5.78, -7.13, -6.72, -9.60, -4.83, -3.44, -4.45, -4.24, -8.30, -7.02]).reshape((K, T))
-
-    n = K  # number of binary dec vars x0 (and z)
-    m = K * T  # number of continuous dec vars u
-    # size of x1 is then n+m
-
-    # Parameters of the Quadratic Programming problem in the standard form for ADMM.
-    ## Objective function
-    Q0 = np.zeros((n, n))
-    c0 = S.reshape(n)
-
-    A0 = np.zeros((1, n))  # we set here one dummy equality constraint
-    b0 = np.zeros(1)  # we set here one dummy equality constraint
-
-    Q1 = np.zeros((m, m))
-    c1 = C.reshape(m)
-
-    ## Constraints
-    # we set here one dummy A1 x0 \leq b1 constraint
-    A1 = np.zeros((1, n))
-    b1 = np.zeros(1)
-
-    # A_2 z + A_3 u \leq b_2 -- > - y_k <= x_{k,t}
-    A2 = np.zeros((m, n))
-    for i in range(K):
-        A2[T * i:T * (i + 1), i] = - np.ones(T)
-    A3 = np.eye(m)
-    b2 = np.zeros(m)
-
-    # A_4 u <= b_3 --> sum_k sum_t D_{kt} x_{kt} \leq P, -x_{kt} \leq 0
-    block1 = D.reshape((1, m))
-    block2 = -np.eye(m)
-    A4 = np.block([[block1], [block2]])
-    b3 = np.hstack([P, np.zeros(m)])
-
-    __dump_arrays(Q0, Q1, c0, c1, A0, b0, A1, b1, A4, b3, A2, A3, b2)
-
-    return K, T, P, S, D, C
+class TestADMMOptimizerMiskp(QiskitOptimizationTestCase):
+    """ADMM Optimizer Tests based on Mixed-Integer Setup Knapsack Problem"""
 
+    def setUp(self):
+        super().setUp()
 
+    def test_admm_optimizer_miskp_eigen(self):
+        """ ADMM Optimizer Test based on Mixed-Integer Setup Knapsack Problem using NumPy eigen optimizer"""
+        K, T, P, S, D, C = self.get_problem_params()
+        miskp = Miskp(K, T, P, S, D, C)
+        op: OptimizationProblem = miskp.create_problem()
 
-class Miskp:
+        # use numpy exact diagonalization
+        qubo_optimizer = MinimumEigenOptimizer(NumPyMinimumEigensolver())
+        continuous_optimizer = CplexOptimizer()
+
+        admm_params = ADMMParameters(qubo_optimizer=qubo_optimizer, continuous_optimizer=continuous_optimizer)
+
+        solver = ADMMOptimizer(params=admm_params)
+        solution = solver.solve(op)
+
+        # debug
+        print("results")
+        print("x={}".format(solution.x))
+        print("fval={}".format(solution.fval))
+
+        correct_solution = [0.009127, 0.009127, 0.009127, 0.009127, 0.009127, 0.009127, 0.009127, 0.009127,
+                            0.009127, 0.009127, 0.006151, 0.006151, 0.006151, 0.006151, 0.006151, 0.006151,
+                            0.006151, 0.006151, 0.006151, 0.006151, 0.,       0.]
+        correct_objective = -1.2113693
+
+        np.testing.assert_almost_equal(correct_solution, solution.x, 3)
+        np.testing.assert_almost_equal(solution.fval, correct_objective, 3)
+
+    def get_problem_params(self):
+        """
+        Fills in parameters for a Mixed Integer Setup Knapsack Problem (MISKP) instance.
+        """
+        #
+
```

Comment:
These names will be invalid, could you give them proper names? Or, since this is just a test file, I think it's okay to add a global `#pylint: disable=invalid-name` at the top of the file if you'd prefer that.

Reply:
Don't get what names are not proper names. Anyway, there are a lot of renamings in the recent code.

Reply:
The variable names must match the regular expression
```
[a-z_][a-z0-9_]{2,30}
```
which means 
```
first character is either a lowercase letter or '_'
  v
[a-z_][a-z0-9_]{2,30}
         ^       ^ 
         |    preceding character expression repeated 2-30 times
    character may be lowercase letter, a number or '_'
```
Practically this means the variable names must have at least 3 letters (and 31 at most), cannot start with a number, must be lowercase and only the special character `_` is allowed.

Reply:
This is clear enough, but 1) I don't see variable names here you are referreing to 2) sometimes you need bloody short names, otherwise there will be names like `var_*` and this is even worse, e.g. you solve an equation for `x`, how do you name `x` then?

Reply:
I was referring to `K`, `T`, etc. which you had before, already changed now (so the comment is resolved) 😄 

Some variables (like `x`, `i` or `j`) are allowed because they are commonly used. And we can always choose to ignore these rules by adding a `#pylint: disable=invalid-name` comment in the code. 

I didn't create these rules and I'm not saying they are flawless! But I believe they help to keep a good level of clarity in the general code and help these people that try to understand the code variables 🙂